client/src/components/bpmn.modeler.component.jsx

```diff
@@ -1,69 +1,104 @@
+import propertiesProviderModule from 'bpmn-js-properties-panel/lib/provider/camunda';
+import camundaModdleDescriptor from 'camunda-bpmn-moddle/resources/camunda';
+const parser = require('../parser.js');
+var convert = require('xml-js');
 
 class BpmnModelerComponent extends Component {
-    modeler = null;
+  modeler = null;
+  componentDidMount = () => {
+    this.modeler = new BpmnModeler({
+      container: '#bpmnview',
+      keyboard: {
+        bindTo: window,
+      },
+      propertiesPanel: {
+        parent: '#propview',
+      },
+      additionalModules: [propertiesPanelModule, propertiesProviderModule],
+      moddleExtensions: {
+        camunda: camundaModdleDescriptor,
+      },
+    });
 
-    componentDidMount = () => {
-        this.modeler = new BpmnModeler({
-            container: "#bpmnview",
-            keyboard: {
-                bindTo: window,
-            },
-            propertiesPanel: {
-                parent: "#propview",
-            },
-            additionalModules: [propertiesPanelModule, propertiesProviderModule],
-            moddleExtensions: {
-                camunda: camundaModdleDescriptor,
-            },
-        });
+    this.newBpmnDiagram();
+  };
 
-        this.newBpmnDiagram();
-    };
+  newBpmnDiagram = () => {
+    this.openBpmnDiagram(emptyBpmn);
+  };
 
-    newBpmnDiagram = () => {
-        this.openBpmnDiagram(emptyBpmn);
-    };
+  openBpmnDiagram = (xml) => {
+    this.modeler.importXML(xml, (error) => {
+      if (error) {
+        return console.log('fail import xml');
+      }
 
-    openBpmnDiagram = (xml) => {
-        this.modeler.importXML(xml, (error) => {
-            if (error) {
-                return console.log("fail import xml");
-            }
+      var canvas = this.modeler.get('canvas');
 
-            var canvas = this.modeler.get("canvas");
+      canvas.zoom('fit-viewport');
+    });
+  };
 
-            canvas.zoom("fit-viewport");
-        });
-    };
+  downloadString = (text, fileType, fileName) => {
+    var blob = new Blob([text], { type: fileType });
 
-    render = () => {
-        return (
-            <div id="bpmncontainer">
-                <div
-                    id="propview"
-                    style={{
-                        width: "25%",
-                        height: "98vh",
-                        float: "right",
-                        maxHeight: "98vh",
-                        overflowX: "auto",
-                    }}
-                ></div>
-                <div
-                    id="bpmnview"
-                    style={{width: "75%", height: "98vh", float: "left"}}
-                ></div>
-            </div>
-        );
-    };
+    var a = document.createElement('a');
+    a.download = fileName;
+    a.href = URL.createObjectURL(blob);
+    a.dataset.downloadurl = [fileType, a.download, a.href].join(':');
+    a.style.display = 'none';
+    document.body.appendChild(a);
+    a.click();
```

Comment:
Are you sure that a is a speaking variable name?

Reply:
Good point. I just replaced a with element :)